test/ostream_joiner_test.cpp

```diff
@@ -6,56 +6,57 @@ Distributed under the Boost Software License, Version 1.0.
 (http://www.boost.org/LICENSE_1_0.txt)
 */
 #include <boost/io/ostream_joiner.hpp>
-#include <boost/core/is_same.hpp>
+#include <boost/type_traits/is_same.hpp>
 #include <boost/core/lightweight_test.hpp>
+#include <boost/core/lightweight_test_trait.hpp>
 #include <sstream>
 
 void test_char_type()
 {
-    BOOST_TEST((boost::core::is_same<char,
-        boost::io::ostream_joiner<const char*>::char_type>::value));
+    BOOST_TEST_TRAIT_TRUE((boost::is_same<char,
```

Comment:
There was a reason I didn't use `BOOST_TEST_TRAIT_TRUE`  because I use <warnings>pedantic and C++03 doesn't like it.

Also, we will need to add  `type_traits` to the CI configurations, or they will fail. They deliberately don't use `depinst` because I want to be aware of what dependencies I have, even for testing.

Reply:
The tests don't generate any warnings in C++03 on my machine. I've updated the CI configs, let's see how it goes.

Reply:
As I see in aa70414fde0c3b337dcaa849b61241efd581a5f4 I explicitly switched away from all `lightweight_test_trait.hpp` when I enabled pedantic, so I'd like to keep it as it is.

Reply:
You used `BOOST_TEST_TRAIT_SAME` back then, not `BOOST_TEST_TRAIT_TRUE`. The latter does not use variadic macros and is strictly C++03 compliant. You can see the latest CI does not produce warnings. Do you really want to revert back to `BOOST_TEST`? Please confirm.

Reply:
Yes, please. Let this change only be about switching from Core to TypeTraits for is_same. (Thanks)

Reply:
Ok, done.